e2e test harness - use busybox from dockerhub

dims opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:

Use the same pattern used everywhere in the e2e test
harness, just use "busybox" (from dockerhub) instead
of using the one from k8s.gcr.io registry.

Change-Id: I700b59f73fd31f2ed6d7f995cd9441839857dd44

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
NONE
```

dims commented:
/assign @ixdy 
/assign @mkumatag

dims commented:
related to https://github.com/kubernetes/kubernetes/issues/66986

mkumatag commented:
/lgtm

neolit123 commented:
/sig testing

dims commented:
/retest

dims commented:
/assign @spiffxp

spiffxp commented:
Honestly I think I'd rather see the magic "busybox" constant get refactored out rather than put another one in.

If this _is_ the correct pattern to use going forward, it should hit test/e2e_node as well (image_id_test, image_list)

dims commented:
ack @spiffxp 

/hold

dims committed:
e2e test harness - use busybox from dockerhub

Use the same pattern everywhere in the e2e test
harness, use busybox (from dockerhub) instead
of using the one from k8s.gcr.io registry.

Change-Id: I57c3b867408c1f9478a8909c26744ea0368ff003

spiffxp commented:
/retest

dims commented:
/hold cancel

dims commented:
@spiffxp @mkumatag - this looks ready now.

mkumatag commented on test/e2e_node/image_id_test.go:
```diff
@@ -28,7 +29,7 @@ import (
 var _ = framework.KubeDescribe("ImageID [NodeFeature: ImageID]", func() {
 
-	busyBoxImage := "k8s.gcr.io/busybox@sha256:4bdd623e848417d96127e16037743f0cd8b528c026e9175e22a84f639eca58ff"
+	busyBoxImage := imageutils.GetE2EImage(imageutils.BusyBox)
```
This test is meant for testing the docker image with sha id instead of numeric tag, so referring the imagename without sha tag will spoil this testcase.

dims replied:
right. fixing.

mkumatag commented:
PR talks about replacing the busybox with docker hub image but I don't see you changing the reference in https://github.com/kubernetes/kubernetes/blob/master/test/utils/image/manifest.go#L55.

dims commented:
@mkumatag right, sorry got lost in the shuffle

spiffxp commented:
/lgtm
/approve
The only "busybox" images I see are in integration now, nice work

mkumatag commented on test/e2e_node/image_list.go:
```diff
@@ -45,7 +45,6 @@ var NodeImageWhiteList = sets.NewString(
 	"google/cadvisor:latest",
 	"k8s.gcr.io/stress:v1",
 	busyboxImage,
-	"k8s.gcr.io/busybox@sha256:4bdd623e848417d96127e16037743f0cd8b528c026e9175e22a84f639eca58ff",
```
why don't have to delete this one here!

dims replied:
oops

mkumatag commented:
/hold

mkumatag commented:
/lgtm
